crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1271,68 +1193,103 @@ where
-                    })
-                },
-                // A long timeout before we hear back from a quorum
-                self.timeouts.pre_quorum_timeout,
-            )
-            .await.map_err(|state| {
+                    }
+                })
+            },
+            // A long timeout before we hear back from a quorum
+            self.timeouts.pre_quorum_timeout,
+        )
+        .await
+        .map_err(|state| {
+            debug!(
+                ?tx_digest,
+                num_unique_effects = state.effects_map.unique_key_count(),
+                bad_stake = state.bad_stake,
+                "Received effects responses from validators"
+            );
+            QuorumExecuteCertificateError {
+                total_stake: self.committee.total_votes,
+                errors: state.errors,
+            }
+        })
+    }
+
+    fn handle_process_certificate_response(
+        &self,
+        tx_digest: &TransactionDigest,
+        state: &mut ProcessCertificateState,
+        response: SuiResult<VerifiedHandleCertificateResponse>,
+        name: AuthorityName,
+    ) -> SuiResult<Option<VerifiedCertifiedTransactionEffects>> {
+        match response {
+            Ok(VerifiedHandleCertificateResponse { signed_effects }) => {
                 debug!(
                     ?tx_digest,
-                    num_unique_effects = state.effects_map.unique_key_count(),
-                    bad_stake = state.bad_stake,
-                    "Received effects responses from validators"
+                    name = ?name.concise(),
+                    "Validator handled certificate successfully",
                 );
-                QuorumExecuteCertificateError {
-                    total_stake: self.committee.total_votes,
-                    errors: state.errors,
+                let signed_effects = signed_effects.into_inner();
+                // Note: here we aggregate votes by the hash of the effects structure
+                match state.effects_map.insert(
+                    (signed_effects.epoch(), *signed_effects.digest()),
+                    signed_effects.data(),
+                    signed_effects.auth_sig().clone(),
+                ) {
+                    InsertResult::NotEnoughVotes => Ok(None),
+                    InsertResult::Failed { error } => Err(error),
+                    InsertResult::QuorumReached(cert_sig) => {
+                        let ct = CertifiedTransactionEffects::new_from_data_and_sig(
+                            signed_effects.into_data(),
+                            cert_sig,
+                        );
+                        match ct.verify(&self.committee) {
+                            Ok(ct) => {
+                                debug!(?tx_digest, "Got quorum for validators handle_certificate.");
+                                Ok(Some(ct))
+                            }
+                            Err(err) => Err(err),
+                        }
```

Comment:
nit:
```suggestion
                        ct.verify(&self.committee).tap_ok(|_| {debug!(?tx_digest, "Got quorum for validators handle_certificate.");})
```

or just have the log in the callsite, just like the error case (line 1219)